### PR TITLE
fix kmeans to respect group_by

### PR DESCRIPTION
**Pull request by double-y**
### Description
fix kmeans to respect group_by

### Checklist

Make sure you have performed following items before submitting this pull request.
If not, please describe the reason.  

- [x] Add test cases for this fix/enhancement
- [x] Pass devtools::check()
- [x] Pass devtools::test()
- [ ] Test installing from github
- [x] Tested with Exploratory

### CODE REVIEW

File: R/broom_wrapper.R [hunk 1]
```diff
@@ -42,7 +42,7 @@ augment_kmeans <- function(df, model, data){
     augment_func <- get("augment", asNamespace("broom"))
     ret <- do.call(augment_func, list(df, model_col, data=data_col))
     # cluster column is factor labeled "1", "2"..., so convert it to integer to avoid confusion
```
Comment (hideaki) [inline R/broom_wrapper.R]: We keep it as factor. So this comment should be modified to reflect that.

Comment (double-y) [inline R/broom_wrapper.R]: fixed it

File: R/model_builder.R [hunk 2]
```diff
@@ -201,8 +192,8 @@ build_kmeans.cols <- function(df, ...,
     if(augment){
       ret <- broom::augment(kmeans_ret, df)
       colnames(ret)[[ncol(ret)]] <- avoid_conflict(grouped_column, "cluster")
-      # cluster column is factor labeled "1", "2"..., so convert it to integer to avoid confusion
-      ret[[ncol(ret)]] <- as.integer(ret[[ncol(ret)]])
+      # make the clustering labels factor with levels with sequence of numbers
+      ret[[ncol(ret)]] <- factor(ret[[ncol(ret)]], levels=seq(centers))
```
Comment (hideaki) [inline R/model_builder.R]: Isn't ret[[ncol(ret)]] already a factor according to the code before change?
If so do we need to apply factor() again?

File: R/model_builder.R [hunk 3]
```diff
@@ -169,13 +158,15 @@ build_kmeans.cols <- function(df, ...,
   }
 
   build_kmeans_each <- function(df){
+    # remove grouping column
```
Comment (hideaki) [inline R/model_builder.R]: why do we need to remove grouping column?

File: R/model_builder.R [hunk 4]
```diff
@@ -102,7 +97,7 @@ build_kmeans.kv_ <- function(df,
     if(augment){
       cluster_column <- avoid_conflict(grouped_column, "cluster")
       row_fact <- as.factor(df[[row_col]])
-      df[[cluster_column]] <- kmeans_ret$cluster[row_fact]
+      df[[cluster_column]] <- factor(kmeans_ret$cluster[row_fact], levels = seq(centers))
```
Comment (hideaki) [inline R/model_builder.R]: Could you add comment on what this line is doing?

File: R/model_builder.R (hunk 2 above)
Comment (double-y) [inline R/model_builder.R]: Right, will remove it

File: R/model_builder.R [hunk 5]
```diff
@@ -77,6 +70,8 @@ build_kmeans.kv_ <- function(df,
   }
 
   build_kmeans_each <- function(df){
+    # remove grouping column
```
Comment (hideaki) [inline R/model_builder.R]: why do we need to remove grouping column here?

File: R/model_builder.R (hunk 3 above)
Comment (double-y) [inline R/model_builder.R]: I'll add comments

File: R/model_builder.R (hunk 5 above)
Comment (double-y) [inline R/model_builder.R]: Added comments

File: R/model_builder.R (hunk 4 above)
Comment (double-y) [inline R/model_builder.R]: Added a comment

**Pull request merged**